ecosystem/platform/server/app/models/user.rb

```diff
@@ -115,6 +115,14 @@ def add_oauth_authorization(data)
     authorizations.build(auth)
   end
 
+  def registration_completed?
+    confirmed? && username_exists?
+  end
+
+  def username_exists?
```

Comment:
Would prefer to just use `username.present?` instead of this method that only works for username.